python/pyspark/sql/connect/session.py

```diff
@@ -287,7 +287,17 @@ def _set_default_and_active_session(cls, session: "SparkSession") -> None:
 
     @classmethod
     def getActiveSession(cls) -> Optional["SparkSession"]:
-        return getattr(cls._active_session, "session", None)
+        s = getattr(cls._active_session, "session", None)
+        if s is not None and not s.is_stopped:
+            return s
+        return None
+
+    @classmethod
+    def getDefaultSession(cls) -> Optional["SparkSession"]:
```

Comment:
Can we avoid adding this as an API in this PR? If we want this, we should add it into `pyspark.sql.session.py` too

Reply:
I've converted this into a private API: `_getDefaultSession()`